tests/test_simplify.py

```diff
@@ -0,0 +1,114 @@
+"""
+This file is part of OSM-flex.
+Copyright (C) 2023 OSM-flex contributors listed in AUTHORS.
+OSM-flex is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+OSM-flex is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+-----
+test simplification functions
+"""
+
+import unittest
+import geopandas as gpd
+import shapely as sh
+from osm_flex.simplify import (remove_small_polygons, remove_contained_points,
+                               remove_contained_polys, remove_exact_duplicates)
+from pathlib import Path
+
+PATH_TEST_DATA = Path(__file__).parent / 'data'
+OSM_FILE = PATH_TEST_DATA / 'test.osm.pbf'
+
+
+# Define some shapes
+coords1 = ((0., 0.), (0., 1.), (1., 1.), (1., 0.), (0., 0.)) #area=1
+coords2 = ((0., 0.), (0., 0.1), (0.1, 0.1), (0.1, 0.), (0., 0.)) #area=0.01
+polygon1 = sh.Polygon(coords1)
+polygon2 = sh.Polygon(coords2)
+point1 = sh.Point(1.0, -1.0)
+point2 = sh.Point(.5, .5)
+line1 = sh.LineString([[0, 0], [1, 0], [1, 1]])
+line2 = sh.LineString([[0.25, 0.25], [0.75, 0.75]])
+
+class TestSimplificationFunctions(unittest.TestCase):
+    def test_remove_small_polygons(self):
+        """ test function remove_small_polygons() """
+        
+        gdf_small_poly = gpd.GeoDataFrame(
+            geometry = [
+                point1,
+                line1,
+                polygon1,
+                polygon2
+            ])
+        
+        gdf_no_small_poly = gpd.GeoDataFrame(
+            geometry = [
+                point1,
+                line1,
+                polygon1,
+            ])
+        
+        # get rid of polygon2
+        gdf_removed = remove_small_polygons(gdf_small_poly, 0.5)
+        
+        self.assertTrue((gdf_removed == gdf_no_small_poly).all().values[0])
```

Comment:
Comparing (Geo)DataFrames can be done more elegantly with [pandas.testing.assert_frame_equal](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.testing.assert_frame_equal.html#pandas.testing.assert_frame_equal)
```python
import pandas.testing as pdt

pdt.assert_frame_equal(gdf_removed, gdf_no_small_poly)
```
(same for all tests in this module)